packages/frint-cli/commands/init.js

```diff
@@ -37,17 +39,27 @@ module.exports = createApp({
           const example = deps.params.example || 'counter';
           const dir = deps.pwd;
 
-          const cmds = [
-            `mkdir -p ${dir}`,
-            `curl https://codeload.github.com/Travix-International/frint/tar.gz/master | tar -xz -C ${dir} --strip=3 frint-master/examples/${example}`,
-          ];
+          function streamFrintExampleToDir() {
+            request('https://codeload.github.com/Travix-International/frint/tar.gz/master')
+              .on('error', deps.console.error)
+              .pipe(tar.x({
+                filter: path => path.indexOf(`frint-master/examples/${example}/`) === 0,
+                strip: 3,
+                C: dir
+              }))
+              .on('error', deps.console.error)
```

Comment:
duplicated handler

Reply:
Since this is based on Node Stream API, the error handling works differently than in promises. The `pipe` operator does not forward errors, so they need to be handled at every step. ([ref](http://grokbase.com/t/gg/nodejs/12bwd4zm4x/should-stream-pipe-forward-errors))

I'm all ears for a more elegant solution :)

Reply:
I could suggest [Pump](https://www.npmjs.com/package/pump) based on this [issue](https://github.com/nodejs/readable-stream/issues/283)

Reply:
Do you still have code duplication in mind or are you talking about cleaning up internal resource of the stream based on the issue you posted?

Reply:
I was mostly thinking about the code duplication and having one error handler. But maybe not required here